tika-parsers/tika-parsers-classic/tika-parsers-classic-modules/tika-parser-zip-commons/src/main/java/org/apache/tika/detect/zip/DefaultZipContainerDetector.java

```diff
@@ -241,12 +247,33 @@ private MediaType detect(ZipArchiveEntry zae, ZipArchiveInputStream zis,
     }
 
     private MediaType finalDetect(StreamingDetectContext detectContext) {
-        for (ZipContainerDetector d : zipDetectors) {
+        for (ZipContainerDetector d : getDetectors()) {
             MediaType mt = d.streamingDetectFinal(detectContext);
             if (mt != null) {
                 return mt;
             }
         }
         return MediaType.APPLICATION_ZIP;
     }
+
+    private List<ZipContainerDetector> getDetectors() {
+        List<ZipContainerDetector> zipDetectors = new ArrayList<>();
+        ServiceLoader loader = getServiceLoader();
+        zipDetectors.addAll(loader.loadDynamicServiceProviders(ZipContainerDetector.class));
+        if (staticZipDetectors0 == null) {
+            staticZipDetectors0 = new ArrayList<ZipContainerDetector>();
+            staticZipDetectors0.addAll(loader.loadStaticServiceProviders(ZipContainerDetector.class));
+        }
+        
+        zipDetectors.addAll(staticZipDetectors0);
```

Comment:
Isn't the concept in Tika, that the passed services are the 'static' services?
So here we only need to add the dynamic services?

Reply:
I cannot follow? If we do not add the staticZipDetectors0 here this will not be used. getDetectors() is supposed to return all detectors

Reply:
> Why do you commit deletion of 'droste.zip' file?
I will add it back as the last commit - it complex to keep track of this external deletion.